### ISSUE TITLE
Issue when exporting default anonymous function or object w/ types

### ISSUE THREAD

**Issue by o0Djeen0o** [r-murphy/babel-plugin-transform-modules-ui5#20]
Hello,

I have a strange behavior with this preset in conjunction with Typescript when generating default export for functions.

if a function is exported as default in a file : 

```
export default (date: Date): string => {
    return "";
};
```
then output in js is incorrect with types remaining where they should be gone

```
sap.ui.define([], function () {

  const __exports = (date: Date): string => {
    return "";
  };

  return __exports;
});
```

If method is not default, for instance : 

```
export function test(date: Date): string {
    return "";
}
```

then output is valid

```
sap.ui.define([], function () {
  function test(date) {
    return "";
  }

  __exports.test = test;
  return __exports;
});
```

I'm using preset version `7.0.0-rc.7` with `@babel/core: 7.3.3` and `@babel/preset-typescript: 7.3.3`

.babelrc looks like : 

```
{
    "ignore": [
        "**/*.d.ts"
    ],
    "presets": [
        "transform-ui5",
        "@babel/preset-typescript"
    ],
    "plugins": [
        [
            "@babel/plugin-syntax-decorators",
            {
                "legacy": true
            }
        ]
    ]
}
```

And if I remove `transform-ui5` from presets list, default function has no more types.

Hope this issue is clear enough. I will test with latest version of the preset later today (rc8)

**Comment by r-murphy** [comment]
Thanks for reporting. I'm able to reproduce and am working on a fix.

Currently, if it the default export is anonymous, the declaration gets saved and `const __exports = {}` is added at the end of the block. But doing so defers adding it to the AST until after types are removed. So I think I'll have to inline it right away so the typescript plugin can still operate on it.

The issue also applied to anonymous objects that have types:

```ts
export default {
  x: 1 as any,
} as any;
```

Produces:

```js
sap.ui.define([], function () {
  return ({
    x: (1 as any)
  } as any);
});
```

**Comment by r-murphy** [comment]
Included in 7.0.0-rc.9. Let me know if you see any other issues with it.
